misk/src/main/kotlin/misk/concurrent/ExecutorServiceModule.kt

```diff
@@ -0,0 +1,34 @@
+package misk.concurrent
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import misk.inject.KAbstractModule
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import kotlin.reflect.KClass
+
+/**
+ */
+class ExecutorServiceModule(
+  private val annotation: KClass<out Annotation>,
+  private val executorService: ExecutorService
+) : KAbstractModule() {
+  override fun configure() {
+    bind<ExecutorService>().annotatedWith(annotation.java).toInstance(executorService)
+  }
+
+  companion object {
+    fun of(annotation: KClass<out Annotation>, nameFormat: String): ExecutorServiceModule {
+      val threadFactory = ThreadFactoryBuilder().setNameFormat(nameFormat).build()
+      return ExecutorServiceModule(annotation, Executors.newCachedThreadPool(threadFactory))
+    }
+
+    fun of(annotation: KClass<out Annotation>, nameFormat: String, nThreads: Int)
```

Comment:
We should follow the kotlin style guide
```
fun of(annotation: KClass<out Annotation>, nameFormat: String, nThreads: Int)
: ExecutorServiceModule {
```

should be
```

fun of(
    annotation: KClass<out Annotation>, 
    nameFormat: String, 
    nThreads: Int
): ExecutorServiceModule {
```

Reply:
done :)